File: native/android/app/src/main/java/org/membercovenantpath/viewer/ui/components/GoldenHourRows.kt
package org.membercovenantpath.viewer.ui.components

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.CheckCircle
import androidx.compose.material.icons.filled.RadioButtonUnchecked
import androidx.compose.material.icons.filled.WarningAmber
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import org.membercovenantpath.viewer.logic.Milestone
import org.membercovenantpath.viewer.logic.Milestones
import org.membercovenantpath.viewer.model.Member
import java.time.LocalDate

/**
 * Golden Hour completion — the user-chosen layout: ONE ROW PER ITEM with a status icon.
 *   ✓ done (green check)   ○ not done (outline circle)   ⚠ data issue (amber warning)
 * N/A items are OMITTED entirely (not eligible — handled upstream in `Milestones.goldenHourRows`), so
 * this view never renders a "N/A" row. The ⚠ row is visibly distinct from both done and not-done.
 *
 * Ported 1:1 from web `components/GoldenHourRows.tsx`. Used by the per-person Baptisms card and the
 * friend/investigator detail to show each Golden Hour step's status at a glance.
 */
private val GREEN = Color(0xFF2E7D32)
private val AMBER = Color(0xFFF9A825)

@Composable
fun GoldenHourRows(
    member: Member,
    list: List<Milestone> = Milestones.all,
    today: LocalDate = LocalDate.now(),
    modifier: Modifier = Modifier,
) {
    val rows = Milestones.goldenHourRows(member, list, today)
    if (rows.isEmpty()) {
        Text(
            "No Golden Hour steps apply yet.",
            style = MaterialTheme.typography.bodySmall,
            color = MaterialTheme.colorScheme.onSurfaceVariant,
            modifier = modifier,
        )
        return
    }
    Column(modifier.fillMaxWidth(), verticalArrangement = Arrangement.spacedBy(2.dp)) {
        rows.forEach { GoldenHourRow(it) }
    }
}

@Composable
private fun GoldenHourRow(row: Milestones.GhRow) {
    val (icon, color, label) = when (row.status) {
        Milestones.GhRowStatus.DONE -> Triple(Icons.Filled.CheckCircle, GREEN, "Done")
        Milestones.GhRowStatus.ISSUE -> Triple(Icons.Filled.WarningAmber, AMBER, "Needs data")
        Milestones.GhRowStatus.NOT_DONE ->
            Triple(Icons.Filled.RadioButtonUnchecked, MaterialTheme.colorScheme.onSurfaceVariant, "Not yet")
    }
    Row(
        Modifier.fillMaxWidth().padding(vertical = 4.dp),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        Icon(icon, contentDescription = label, tint = color, modifier = Modifier.size(18.dp))
        Spacer(Modifier.width(10.dp))
        Text(
            row.milestone.label,
            style = MaterialTheme.typography.bodyMedium,
            modifier = Modifier.weight(1f),
        )
        Text(label, color = color, style = MaterialTheme.typography.bodySmall, fontWeight = FontWeight.Medium)
    }
}

File: native/ios/Sources/CovenantPathKit/Views/Shared/GoldenHourRows.swift
#if canImport(UIKit)
import SwiftUI

/// Golden Hour completion — the user-chosen layout: ONE ROW PER ITEM with a status icon.
///   ✓ done (green check)   ○ not done (outline circle)   ⚠ data issue (amber warning)
/// N/A items are OMITTED entirely (not eligible — handled upstream in `Milestones.goldenHourRows`), so
/// this view never renders an "N/A" row. The ⚠ row is visibly distinct from both done and not-done.
///
/// Faithful port of web `apps/web/src/components/GoldenHourRows.tsx`. Used by the person/friend detail
/// view (and the per-person Baptisms card) to show each Golden Hour step's status at a glance.
struct GoldenHourRows: View {
    let member: Member
    /// `list` lets a caller include the longer-horizon covenants (pass `Milestones.needsCategories`);
    /// defaults to the integration milestones.
    var list: [Milestone] = Milestones.all

    private static let green = Color(hex: 0x2E7D32)
    private static let amber = Color(hex: 0xF9A825)

    var body: some View {
        let rows = Milestones.goldenHourRows(member, list: list)
        if rows.isEmpty {
            Text("No Golden Hour steps apply yet.").font(.callout).foregroundStyle(.secondary)
        } else {
            VStack(alignment: .leading, spacing: 8) {
                ForEach(rows) { row in
                    GoldenHourRowView(row: row)
                }
            }
        }
    }
}

private struct GoldenHourRowView: View {
    let row: Milestones.GhRow

    var body: some View {
        let v = visuals(row.status)
        HStack(spacing: 10) {
            Image(systemName: v.icon)
                .font(.system(size: 17))
                .foregroundStyle(v.color)
                .accessibilityLabel(v.accessibility)
            Text(row.milestone.label)
                .font(.callout)
            Spacer(minLength: 8)
            Text(v.label)
                .font(.caption.weight(.medium))
                .foregroundStyle(v.color)
        }
        // The full description from the single milestone definition is the tooltip equivalent.
        .accessibilityHint(row.milestone.label)
    }

    private func visuals(_ status: Milestones.GhRowStatus)
        -> (icon: String, color: Color, label: String, accessibility: String) {
        switch status {
        case .done:
            return ("checkmark.circle.fill", Color(hex: 0x2E7D32), "Done", "Completed")
        case .issue:
            // A DATA ISSUE — the value should be known but isn't (sentinel / missing). Distinct from
            // N/A (omitted) and from an honest not-done.
            return ("exclamationmark.triangle.fill", Color(hex: 0xF9A825), "Needs data",
                    "Data issue — value unavailable")
        case .notDone:
            return ("circle", Color(.systemGray), "Not yet", "Not yet completed")
        }
    }
}
#endif
